test_conformance/api/test_api_consistency.cpp

```diff
@@ -473,14 +437,689 @@ int test_consistency_pipes(cl_device_id deviceID, cl_context context,
+        test_failure_error(error, CL_INVALID_OPERATION,
+                           "Device does not support Creating a 2D Image from a "
+                           "Buffer but clCreateImageWithProperties did not "
+                           "return CL_INVALID_OPERATION");
+    }
+    else
+    {
+        test_assert_error(supports_cl_khr_image2d_from_buffer,
+                          "Device supports Creating a 2D Image from a Buffer "
+                          "but does not support cl_khr_image2d_from_buffer");
+    }
+
+    return TEST_PASS;
+}
+
+// Nothing needed for sRGB Images:
+// All of the sRGB Image Channel Orders (such as CL_​sRGBA) are optional for
+// devices supporting OpenCL 3.0.
+
+int test_consistency_depth_images(cl_device_id deviceID, cl_context context,
+                                  cl_command_queue queue, int num_elements)
+{
+    // The CL_DEPTH Image Channel Order is optional for devices supporting
+    // OpenCL 3.0.
+    cl_int error;
+
+    cl_uint totalDepthImageFormats = 0;
+
+    const cl_mem_flags mem_flags[] = {
+        CL_MEM_WRITE_ONLY,
+        CL_MEM_READ_WRITE,
+        CL_MEM_KERNEL_READ_AND_WRITE,
+    };
+    for (int i = 0; i < ARRAY_SIZE(mem_flags); i++)
+    {
+        cl_uint numImageFormats = 0;
+        error = clGetSupportedImageFormats(context, mem_flags[i],
+                                           CL_MEM_OBJECT_IMAGE2D, 0, NULL,
+                                           &numImageFormats);
+        test_error(
+            error,
+            "Unable to query number of CL_MEM_OBJECT_IMAGE2D image formats");
+
+        std::vector<cl_image_format> imageFormats(numImageFormats);
+        error = clGetSupportedImageFormats(
+            context, mem_flags[i], CL_MEM_OBJECT_IMAGE2D, imageFormats.size(),
+            imageFormats.data(), NULL);
+        test_error(error,
+                   "Unable to query CL_MEM_OBJECT_IMAGE2D image formats");
+        for (auto& imageFormat : imageFormats)
         {
-            log_error("CL_DEVICE_PIPE_SUPPORT returned CL_TRUE but "
-                      "CL_DEVICE_GENERIC_ADDRESS_SPACE_SUPPORT returned "
-                      "CL_FALSE\n");
-            return TEST_FAIL;
+            if (imageFormat.image_channel_order == CL_DEPTH)
+            {
+                totalDepthImageFormats++;
+            }
         }
     }
 
+    bool supports_cl_khr_depth_images =
+        is_extension_available(deviceID, "cl_khr_depth_images");
+
+    if (totalDepthImageFormats == 0)
+    {
+        test_assert_error(supports_cl_khr_depth_images == false,
+                          "Device does not support Depth Images but does "
+                          "support cl_khr_depth_images");
+    }
+    else
+    {
+        test_assert_error(supports_cl_khr_depth_images,
+                          "Device supports Depth Images but does not support "
+                          "cl_khr_depth_images");
+    }
```

Comment:
A detail: I'm wondering whether "report" wouldn't be better than "support" in the error messages. We're defining support as reporting the extension _and_ at least one format.

Reply:
I don't feel strongly about this.  Do you want this changed?  If so, just tell me what to change it to :).

Reply:
I won't lose sleep over that :).